### PR TITLE
2783 validation decorator i agent configuration repository

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #2783 .

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py
Comment (VakarisZ) [inline monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py:28]: This is not a decorator it's a wrapper, because it applies only to this method. Might as well just call it "get_valid_configuration" and do this in the repository

File: monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py
Comment (VakarisZ) [inline monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py:23]: Can't we just use the schema?

File: monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py:28]: How it is just only applying to this method when we validate the configuration before we update?

File: monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py:23]: Do you mean `AgentConfiguration.schema()`? I don't think so, as the compiler is changing that schema with the plugins.

File: monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py
Comment (mssalvatore) [reply monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py:28]: "Wrapper" is another name for "decorator". Please see [this](https://refactoring.guru/design-patterns/decorator) for a discussion of the decorator pattern.

**Review by shreyamalviya — APPROVED** [review]
This works after rebasing on [2785-plugins-in-schema](https://github.com/guardicore/monkey/tree/2785-plugins-in-schema), but we should rebase and merge this after the work on that branch is merged to `develop`.

File: monkey/tests/unit_tests/monkey_island/cc/repositories/test_agent_configuration_validation_decorator.py
Comment (mssalvatore) [inline monkey/tests/unit_tests/monkey_island/cc/repositories/test_agent_configuration_validation_decorator.py:44]: This looks like it gets duplicated a lot. Can we put these in fixtures?

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py
Comment (mssalvatore) [inline monkey/monkey_island/cc/repositories/agent_configuration_validation_decorator.py:17]: This format is incorrect. It should be:

```
"""
Single-line summary

Multi-line description
"""
```

See https://peps.python.org/pep-0257/#multi-line-docstrings for more details.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**
